web/templates/confirm_dialog/confirm_deactivate_or_reactivate_stream.hbs

```diff
@@ -26,3 +27,29 @@
         This action cannot be undone.
     {{/tr}}
 </p>
+{{#if is_announcement_stream}}
+<p class="notification_stream_archive_warning">{{#tr}}Archiving this channel will also disable settings that were configured to use this channel:{{/tr}}</p>
+<ul>
+    {{#if is_new_stream_announcements_stream}}
+        <li>{{#tr}}New channel notifications{{/tr}}</li>
+    {{/if}}
+    {{#if is_signup_announcements_stream}}
+        <li>{{#tr}}New user notifications{{/tr}}</li>
+    {{/if}}
+    {{#if is_zulip_update_announcements_stream}}
+        <li>{{#tr}}Zulip update announcements{{/tr}}</li>
+    {{/if}}
+</ul>
+{{/if}}
+{{else}}
+<p>
+    {{#tr}}Unarchiving this channel will:{{/tr}}
+</p>
+<p>
+    <ul>
+        <li>{{#tr}}Make it appear in the left sidebar for all subscribers.{{/tr}}</li>
+        <li>{{#tr}}Allow sending new messages to this channel.{{/tr}}</li>
+        <li>{{#tr}}Allow messages in this channel to be edited, deleted, or moved.{{/tr}}</li>
+    </ul>
+</p>
+{{/unless}}
```

Comment:
I think we can have different files for archiving and unarchiving modals. This way we can also make the file name small.